src/main/resources/assets/opencomputers/loot/OpenOS/lib/event.lua

```diff
@@ -118,28 +144,50 @@ end
 
 function event.pull(...)
   local args = table.pack(...)
-  local seconds, name, filter
   if type(args[1]) == "string" then
-    name = args[1]
-    filter = table.pack(table.unpack(args, 2, args.n))
+    return event.pullFiltered(createPlainFilter(...))
   else
     checkArg(1, args[1], "number", "nil")
     checkArg(2, args[2], "string", "nil")
-    seconds = args[1]
-    name = args[2]
-    filter = table.pack(table.unpack(args, 3, args.n))
+    return event.pullFiltered(args[1], createPlainFilter(select(2, ...)))
   end
+end
 
-  local hasFilter = name ~= nil
-  if not hasFilter then
-    for i = 1, filter.n do
-      hasFilter = hasFilter or filter[i] ~= nil
+function event.pullMultiple(...)
+  local seconds
+  local args
+  if type(...) == "number" then
+    seconds = ...
+    args = table.pack(select(2,...))
+    for i=1,args.n do
+      checkArg(i+1, args[i], "string", "nil")
     end
+  else
+    args = table.pack(...)
+    for i=1,args.n do
+      checkArg(i, args[i], "string", "nil")
+    end
+  end
+  return event.pullFiltered(seconds, createMultipleFilter(table.unpack(args, 1, args.n)))
+
+end
+
+function event.pullFiltered(...)
+  local args = table.pack(...)
+  local seconds, filter
+
+  if type(args[1]) == "function" then
+    filter = args[1]
+  else
+    checkArg(1, args[1], "number", "nil")
+    checkArg(2, args[2], "function", "nil")
+    seconds = args[1]
+    filter = args[2]
   end
 
   local deadline = seconds and
                    (computer.uptime() + seconds) or
-                   (hasFilter and math.huge or 0)
+                   (filter and math.huge or 0)
```

Comment:
There's no way of having no filter in here anymore, so that could always be `math.huge`, unless I'm missing something? Better yet, just set seconds to `math.huge` in the non-timeout if branch, I guess?